### ISSUE TITLE
Changing grouping for the PHA data grouped outside Sherpa and grouping the background data

### ISSUE THREAD

**Issue by anetasie** [sherpa/sherpa#1603]
This is a copy from my comment in the PR #1486. 

We have a separate Google Doc with the written requirements for grouping/filtering of the PHA files.
https://docs.google.com/document/d/1PqScMS4teD04vbHbiDHK271Naarq81DbwOouLIgZhz4/edit?usp=sharing

Both the new GDoc document and the old requirement documents the background data is always ungroup and in the case of background subtraction the grouping from the source is applied to the background data. See the last page of this old document:
https://drive.google.com/open?id=1hfEk3qTR_KJQkO6UXXkExWjC7ca7yCTQ

For background fitting only one would want to potentially group the background in a different way than the source data.

The example in this PR shows the data set which has both the source and background data grouped outside sherpa, and both src/bkg have the same groups. However, in 4.14. and in this PRs `group_counts(10)` is applied to both the source and the background resulting in the bin edges, which don't match between src/bkg (see plot).  I realized that the code you are validating is needed for the `subtract()`. It seems that 'ungroup' in this PR has a different result in `ungroup` in 4.14.

These two figures show the original grouping and the modified grouping with `group_counts(10)`. These are the same in this PR and 4.14.

![input_group](https://user-images.githubusercontent.com/1213746/191357540-211b4c64-0318-478d-939a-916e6b249f5b.png)
![new-group-10](https://user-images.githubusercontent.com/1213746/191357657-ecddba63-9fdc-41af-ba5a-3b373ba4574c.png)

Interestingly when I ungroup the background then  I'm no longer able to change the group setting for the background with `set_grouping()` and this is for unfiltered data.

_Originally posted by @anetasie in https://github.com/sherpa/sherpa/issues/1486#issuecomment-1252884568_

**Comment by anetasie** [comment]
@DougBurke yes, I'll try with the 4.15 and post it here.

**Comment by DougBurke** [comment]
@anetasie I don't understand what you want here. Here's an example run with Sherpa 4.15.0:


```
sherpa-4.15.0> load_pha("3c273.pi")
WARNING: systematic errors were not found in file '3c273.pi'
statistical errors were found in file '3c273.pi' 
but not used; to use them, re-read with use_errors=True
read ARF file 3c273.arf
read RMF file 3c273.rmf
WARNING: systematic errors were not found in file '3c273_bg.pi'
statistical errors were found in file '3c273_bg.pi' 
but not used; to use them, re-read with use_errors=True
read background file 3c273_bg.pi

sherpa-4.15.0> plot_data(); plot_bkg(overplot=True)

```

![screenshot1](https://user-images.githubusercontent.com/224638/201707742-e8f84fd3-c623-4bab-9551-90e2415cf929.png)


```
sherpa-4.15.0> group_counts(50)

sherpa-4.15.0> plot_data(); plot_bkg(overplot=True)

```

![screenshot2](https://user-images.githubusercontent.com/224638/201707814-ab477e28-2fdc-4844-a703-23c75cfe0646.png)


```
sherpa-4.15.0> group_counts(10, bkg_id=1)

sherpa-4.15.0> plot_data(); plot_bkg(overplot=True)

```

![screenshot3](https://user-images.githubusercontent.com/224638/201707876-5e7f8f3e-9202-4f06-9b74-c7634e9cc68b.png)


```
sherpa-4.15.0> group_counts(20)

sherpa-4.15.0> plot_data(); plot_bkg(overplot=True)

```

![screenshot4](https://user-images.githubusercontent.com/224638/201707918-33cc4ccb-fcb8-4c35-b3de-d58ee8a13c50.png)

You can see that the `group_counts` call has changed the grouping for the source and background regions. 

There is a subtlety that we do not really describe (as it's hard to say where this should be without repeating it for all the groupxxx calls, which we have done before but is tiring to do) is that if you want to change the source and background regions then you should change the background region second - i.e.

    group_xxx(value1)
    group_xxx(value2, bkg_id=1)

This is because we have no way to say "call group_counts only on the source region for a given dataset"; we automatically run it on both source and background regions.

**Comment by anetasie** [comment]
@DougBurke There are a few cases where you want to compare the number of source and background counts for the groups which cover the same channels - in plotting for visual inspection, and when calculating statistics. For example `wstat` uses the number of counts in a channel directly and for that reason the groups should match. Of course, typically we don't group in this statistics, but based on the earlier discussions the use of the statistics require grouping to have at least 1 count in each channel/group, so the channel numbers in each group should be the same for the source and the background. In your final example the plot illustrates a mismatch between the source and background groups in number of channel per group. How this would be used in calculating the statistics?

The situation is different when fitting the background model to background data `fit_bkg` when one may require a specific binning for the background data  not necessarily matched to the number of channels per bin in the source data.

In your first plotting examples - for the data binned outside Sherpa plots appear to have initially matched groups between source and background, and this is changed when we start regrouping the channels.

**Comment by DougBurke** [comment]
@anetasie  I do not understand what you want and when you want it. We need to work this out before trying to change things or explain what is or isn't happening.

**Comment by DougBurke** [comment]
For reference: when subtracting data, it doesn't matter what `plot_bkg` shows as the source filter/grouping is always used. However, I don't want the plot bkg commands to do this (they could grab the source settings and always use them) as this reduces flexibility and is actually harder than you might think to do (as we don't have a reference from a background dataset to the source dataset). 

Maybe it would be helpful to have a copy_filter_status style routine (bikesheding desperately needed) to copy the settings from the source to background datasets (for both plotting subtracted data and for handling cases where we treat them separately in some sense).

This is why having an idea of what you actually want to do, and how it interacts with other things we can do now, is needed.

**Comment by DougBurke** [comment]
So, I claim that you can have different grouping for data and background, or the same grouping (and if you subtract it doesn't matter what the background grouping). Therefore I think you can have what you want, the questions are

- is this documented well enough
- are there certain settings where we could make things "nicer" for the user
- are we missing some functionality

One thing that XSPEC lets you do is group data differently than how you fit it. We can do this (whether with the same options that XSPEC has is an open question) but it is a bit messy to do (if you want to restore the data to the original state). I'm not sure if this is part of this issue or a separate one.

**Comment by DougBurke** [comment]
@anetasie - is this issue still open or not?

**Comment by DougBurke** [comment]
@anetasie ping

**Comment by DougBurke** [comment]
@anetasie is this issue still open?

**Comment by anetasie** [comment]
This was addressed with #1892
